First version of the Immune to Spells-spell

LudvigStridsberg opened the pull request:
Added a file containing the spell and "required" it in the spell-index-file. The file is partly based on the locks-file.

LudvigStridsberg committed:
First version of the Immune to Spells-spell

Added a file containing the spell and "required" it in the spell-index-file. The file is partly based on the locks-file.

LudvigStridsberg committed:
Updated file name

Updated the name of the file so that it is more consistent with my other ones.

trumtomte commented on spells/immuneToSpells.js:
```diff
@@ -0,0 +1,44 @@
+    context.updateGameState(game);
+
+    // Message the player about their immunity
+    context.send('player:immunity', {duration: '30 seconds'});
```
Change it to be in seconds, this is a string now. So `30 * 1000` or - if you store the duration in a variable somewhere.

trumtomte commented on spells/immuneToSpells.js:
```diff
@@ -0,0 +1,44 @@
+
+        // Message the player that their immunity has ran out
+        context.send('player:immunity:faded', {});
+    }, 30000);
```
I think (based on the comment above) we should have a variable at the start of the function `const duration = 30 * 1000;` for ease of use.

trumtomte reviewed (changes requested):
Some small fixes.

LudvigStridsberg committed:
Second version of Immune to Spells-spell

A requested by Sebastian a constant  that has the duration of the spell as value was implemented and the code was changed so that it uses the constant instead of integers and strings.

Co-Authored-By: Sebastian Bengtegård <sebastianbengtegard@gmail.com>

trumtomte reviewed (approved):
👍🏻

Commit:
Merge branch 'main' into Spell_Immune_to_Spells
